lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -845,20 +845,19 @@ sub save_screenshot($) {
     unlink($tmpdir . "/$current") if $current;
 }
 
-sub append_log($) {
-    my ($self, $log) = @_;
+sub append_log($$) {
+    my ($self, $log, $file_name) = @_;
     return unless length($log->{data});
 
-    my $file = $self->worker->get_property('WORKER_TMPDIR');
-    return unless -d $file;    # we can't help
-    $file .= "/autoinst-log-live.txt";
-    if (sysopen(my $fd, $file, Fcntl::O_WRONLY | Fcntl::O_CREAT)) {
-        sysseek($fd, $log->{offset}, Fcntl::SEEK_SET);
-        syswrite($fd, $log->{data});
+    my $path = $self->worker->get_property('WORKER_TMPDIR');
+    return unless -d $path;    # we can't help
+    $path .= "/$file_name";
+    if (open(my $fd, '>>', $path)) {
```

Comment:
The main change is from using a file offset sent by the worker to simply opening the file for appending. It seems to work fine this way and allows me to easily merge the two serial streams. As far as I know there is no significant difference between `open` and `sysopen` in the this case.